add cluster number to variant naming

singlesp opened the pull request:
Closes .

## Changes proposed in this pull request

Key changes made:

Added a variant_counts dictionary to track the number of variants for each parameter combination
Modified the variant detection loop to:

Collect all variant parameters in a list first
Sort them for consistent ordering
Create a unique key from the parameters
Increment and append the counter for that variant type
Changed the variant naming to include the counter number at the end
Maintained the "Other" variant case with numbering as well

This will now produce variant names like:

VARIANTEchoTime1
VARIANTEchoTime2
VARIANTRepetitionTimeEchoTime1
VARIANTOther1

## Documentation that should be reviewed

-

tsalo commented on cubids/utils.py:
```diff
@@ -769,6 +769,9 @@ def assign_variants(summary, rename_cols):
         The updated summary DataFrame with a new column "RenameEntitySet"
         containing the new entity set names for each file.
     """
+    # Track variant counts for each parameter
+    variant_counts = {}
```
I'd rather use the cluster value (when there is one) than a new iterator

tsalo commented:
@singlesp can you ask Cursor to write a test that would fail before but will succeed with the change? Something that checks the RenameEntitySet column after `cubids group`?

singlesp commented:
This test suite covers:
Basic variant numbering functionality
Multiple entity sets handling
Acquisition entity handling
Fieldmap-related variants
Consistency across multiple runs

The tests use the same structure as the existing tests (see reference to test_apply.py, lines 203-236) and follow similar patterns for setting up test data.

The tests verify that:
Variants of the same type are numbered sequentially
Different entity sets maintain separate numbering
Existing acquisition entities are properly handled
Fieldmap variants are correctly named
The numbering is consistent and reproducible

singlesp commented:
Key changes:

Removed the variant_counts dictionary
Added tracking of cluster values alongside variant parameters
Modified the variant string creation to include cluster numbers when available
Updated tests to verify cluster-based naming
Maintained special handling for fieldmap-related variants which don't use clusters

The variant names will now look like:
VARIANTEchoTime2 (where 2 is the cluster number)
VARIANTEchoTime2RepetitionTime3 (for multiple clustered parameters)
VARIANTNoFmap (for non-clustered parameters)

tsalo commented on cubids/utils.py:
```diff
@@ -795,37 +796,45 @@ def assign_variants(summary, rename_cols):
 
                 if f"Cluster_{col}" in dom_entity_set.keys():
                     if summary.loc[row, f"Cluster_{col}"] != dom_entity_set[f"Cluster_{col}"]:
-                        acq_str += col
```
I could be wrong, but I think it would be simpler to just do

```python
acq_str += f"{col}{int(summary.loc[row, f'Cluster_{col}'])}"

tsalo commented on cubids/tests/test_variant_numbering.py:
```diff
@@ -0,0 +1,149 @@
+import pytest
+from cubids.cubids import CuBIDS
+from cubids.utils import assign_variants
+
+@pytest.fixture
+def sample_summary_df():
```
It only put one empty line even though Python convention is two.
